tasks/main.yml

```diff
@@ -91,25 +91,47 @@
   # Force ansible to know the IP address of this host so it can connect
   # This fixes things if you don't have the IP address in DNS or /etc/hosts.
 - name: Set inventory_hostname to the new guest IP address
-  set_fact: inventory_hostname={{ internal_ip }}
+  set_fact: ansible_host={{ internal_ip }}
 
 - name: wait for VM to become available before continuing
   wait_for:
     port: 22
-    host: "{{ provisioning_wait_for_addr | default(inventory_hostname) }}"
+    host: "{{ provisioning_wait_for_addr | default(ansible_host) }}"
     timeout: 300
+    search_regex: OpenSSH
   delegate_to: "{{ bastion_host | default(hyper) }}"
   when: guest_exists | failed
 
+  # This attempts to handle the common case where the new VM has a new and
+  # unknown host key, because you just rebuilt this VM. Modern linux distro's
+  # protect ~/.ssh linux using selinux so you'll need libselinux-python.
+  # libselinux doesn't work in python virtual envs without screwing about,
+  # see:
+  # https://dmsimard.com/2016/01/08/selinux-python-virtualenv-chroot-and-ansible-dont-play-nice/
+  # Since this is a hassle this task is set to ignore errors. Pull requests
+  # are welcome! ;-)
 - name: Clear known_hosts for new VM
   known_hosts:
     name: "{{ item }}"
     state: absent
   when: guest_exists | failed
-  delegate_to: "{{ bastion_host | default(hyper) }}"
+  delegate_to: localhost
   with_items:
-    - "{{ internal_ip }}"
+    - "{{ ansible_host }}"
     - "{{ inventory_hostname }}"
+  ignore_errors: yes
+
+- name: Disable host key checks since next step will always prompt for a new host key
+  set_fact:
+    ansible_ssh_common_args: "-o StrictHostKeyChecking=no"
```

Comment:
Overriding `ansible_ssh_common_args` could be problematic. The other `set_fact` task was overriding ansible_host which is also a builtin, but that should be fine since it's just one value and the correct value is basically what is set at runtime. However, `ansible_ssh_common_args` can contain a list of options and the user may have set its value to something else already. For example, you can do a jump host configuration using `ansible_ssh_common_args` like this:
https://github.com/CSC-IT-Center-for-Science/etherpad-deployment-demo/blob/master/group_vars/galera.yml#L2

Could we find a different way to solve this problem instead? Maybe through an SSH config file setting instead? So basically when you use Ansible you always have this (maybe not for all hosts but a subset if you like) in your SSH config file:

```
Host *
  StrictHostKeyChecking no
```